### DISCUSSION TITLE
New label

### DISCUSSION THREAD

**Discussion by Elliria** [autokey/autokey#996] [Maintenance goals]
It occurs to me that we should probably have a **pre-release** label or something that's worded better than that to convey tasks that need to be done before the next release. Thoughts?

**Comment by josephj11** [comment]
There must be a best practices recommendation somewhere.
I created a 0.96.1 label to do that. I could  create a 0.97.0 label.

Doing the above would clutter our labels. I forget exactly what we determined before, but maybe in this case, deleting the label after the release wouldn't hurt anything and that would eliminate the clutter.

If I created something like release blocker (or prerelease)... label without being specific to the release, that would work, but we'd have to make sure it was only used on the current next release so that something blocking a subsequent release woudn't get considered now or accidentally cleared when the current version is released and  the labels cleared...

Here's what Venice.ai came up with:

-----

On github, I would like a way to mark the issues and pull requests that have to be completed for the next release of a project. I need to be able to convenietly clear those marks after the release is done. What are best practices for doing this?

On GitHub, you can use labels and milestones to mark issues and pull requests that need to be completed for the next release. Here are some best practices to help you achieve this:

Create a milestone: Create a new milestone for the upcoming release. This will help you track all the issues and pull requests associated with that release. You can add a description to the milestone to include the release version, target date, and any other relevant information.

Use a specific label: Create a label (e.g., "next-release" or "release-candidate") and apply it to all the issues and pull requests that need to be completed for the next release. This label will help you filter and track the relevant issues and pull requests.
Assign issues and pull requests to the milestone: Assign the issues and pull requests with the "next-release" label to the milestone you created. This will help you visualize the progress and track the dependencies.
Use the "next-release" label as a filter: Use the "next-release" label as a filter in the issues and pull requests tabs to quickly see all the items that need to be completed for the next release.
Clear the label after the release: Once the release is done, you can remove the "next-release" label from all the issues and pull requests. This will help you start fresh for the next release.
Additional tips:

You can also use GitHub's Projects feature to create a board for tracking the issues and pull requests for the next release.

Consider using GitHub's release draft feature to create a draft release and associate it with the milestone. This will help you track the progress and create a release notes document.
If you have a large number of issues and pull requests, you can use GitHub's query language to create a custom filter that includes the "next-release" label and other relevant criteria.
Example of how to use GitHub's query language:


Copy code

is:issue is:open label:"next-release" milestone:"Next Release"
This query will show all open issues with the "next-release" label and assigned to the "Next Release" milestone.

By following these best practices, you can efficiently track and manage the issues and pull requests for your next release, and easily clear the marks after the release is done.

-----

The milesones, projects, and draft release features sound interesting. I have never looked into any of them and don't have time now.

**Comment by Elliria** [comment]
Thanks. I've never done a milestone, but [the AutoKey project has](https://github.com/autokey/autokey/milestones) and it looks simple enough to do.

**Comment by Elliria** [comment]
I just added a closed issue to the 0.97.0 milestone. Very cool. Okay, this discussion is no longer needed. Thanks for the help!

**Comment by josephj11** [comment]
Do we still want/need a 0.97.0 label as well?

**Comment by Elliria** [comment]
I think so. It can be used to do searches. I haven't checked whether it's been added to every issue that's part of that milestone. I'll add it to my task list. It will be quick and easy.
